Pin currently playing song for its duration

Cephian opened the pull request:
Closes #63.

It wasn't 100% settled whether this was a good idea or not, however I think it is. It's true that it makes the pinned result of the `!list` command slightly harder to view while another song is playing. I think this is not a huge deal, since

1. Users already have a tracklist they are following along with if they are voting as they go (which voting stats seem to suggest most people do)
2. It takes minimal effort to scroll down slightly and view the tracklist, and the majority of the time the pinned message view is "stable" (e.g. not actively changing from new pins/unpins)
3. Since tracks can have album art and more info, there is a lot to be gained by being able to view them during a bust and not just for a fleeting moment at the start
4. Busty's nick can provide some at-a-glance info, but large collab songs often have long names with critical info cut off, and it can be difficult to find which song is playing at a glance in the list (granted #19 would help with this somewhat)

I consider the pinned tracklist to be more for initial and post-bust reference than mid-bust reference. I recommend that we implement this feature, and then next bust we tell people about it and ask them if we should keep it or remove it.

anoadragon453 reviewed (changes requested):
Tested, and it works! You're right in that one really only occasionally checks pins for the current spot in the overall list of songs. So the chance of pins shifting while someone is viewing them is slim, and even in that case it's not a big deal.

Cephian commented:
Removing the `inner_f` was a good change style-wise but was so grand it made the diffs go wonky (and will probably mandate manual refactoring on most pending PR's), so I'll summarize the changes here manually myself.

1. Printed error if `ffmpeg_post_hook` was invoked with one
2. Added async function `try_set_pin(message, pin_state)` which pins/unpins while handling error and permissions logging. call `try_set_pin(message, True)` to pin it, and `try_set_pin(message, False)` to unpin.
3. removed `inner_f` wrapper function and called play_next_song as async from `ffmpeg_post_hook`.
